@xen-orchestra/rest-api/src/users/user.controller.mts

```diff
@@ -0,0 +1,50 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import { Request as ExRequest } from 'express'
+import { provide } from 'inversify-binding-decorators'
+import { notFoundResp, unauthorizedResp, Unbrand } from '../open-api/common/response.common.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import { XoUser } from '@vates/types'
+import { partialUsers, user, userIds } from '../open-api/oa-examples/user.oa-example.mjs'
+
+@Route('users')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('users')
+@provide(UserController)
+export class UserController extends XoController<XoUser> {
+  // --- abstract methods
+  getAllCollectionObjects(): Promise<XoUser[]> {
+    return this.restApi.xoApp.getAllUsers()
+  }
+  getCollectionObject(id: XoUser['id']): Promise<XoUser> {
+    return this.restApi.xoApp.getUser(id)
+  }
+
+  /**
+   * @example fields "email,id"
+   * @example filter "permission:admin"
+   * @example limit 42
+   */
+  @Example(userIds)
+  @Example(partialUsers)
+  @Get('')
+  async getUsers(
+    @Request() req: ExRequest,
+    @Query() fields?: string,
+    @Query() filter?: string,
+    @Query() limit?: number
+  ): Promise<string[] | WithHref<Partial<Unbrand<XoUser>>>[]> {
+    return this.sendObjects(Object.values(await this.getObjects({ filter, limit })), req)
+  }
+
+  /**
+   * @example id "722d17b9-699b-49d2-8193-be1ac573d3de"
+   */
+  @Example(user)
+  @Get('{id}')
+  @Response(notFoundResp.status, notFoundResp.description)
+  getUser(@Path() id: string): Promise<Unbrand<XoUser>> {
+    return this.getObject(id as XoUser['id'])
+  }
```

Comment:
These methods will return the `pw_hash` of a user.
I don't think it's a good practice to return the password (even if it's a hash).
@fbeauchamp Are you agree to not return the hashed password? (The JSON-RPC API does not return it)

Reply:
I agree, even if it's only a salted hash, it has no value , and only add one more risks (at leats questions by user asking if it's safe)